tests/accelerators/test_ipu.py

```diff
@@ -460,6 +460,45 @@ def test_manual_poptorch_opts_train_grad_accum(tmpdir):
         assert trainer.accelerator.training_type_plugin.inference_opts.Training.gradient_accumulation == 1
 
 
+@RunIf(ipu=True)
+def test_manual_poptorch_opts_custom(tmpdir):
+    """
+    Ensure if the user passes manual poptorch Options with custom parameters set,
+    we respect them in our poptorch options.
+    """
+
+    model = IPUModel()
+    inference_opts = poptorch.Options()
+    inference_opts.deviceIterations(16)
+    inference_opts.replicationFactor(2)
+    inference_opts.Training.gradientAccumulation(1)
+
+    training_opts = poptorch.Options()
+    training_opts.deviceIterations(8)
+    inference_opts.replicationFactor(2)
+    training_opts.Training.gradientAccumulation(2)
+
+    trainer = Trainer(
+        default_root_dir=tmpdir,
+        ipus=2,
```

Comment:
I wonder if we should change this behaviour (of setting ipu=). It's still a bit odd because our manual pipelining/option passing leads to so many ways to run with more/less IPUs than are set here. Do you think maybe we should make this optional and if `IPUPlugin` is set we take the parameters from there?

Reply:
I agree, this is my current thinking:

`ipus=x`  is a nice way for users already familiar with lightning to get started with IPUs, but quickly this can be outgrown to swap to the `IPUPlugin`.

In addition, the IPUPlugin should override the Trainer. I.e setting poptorch.Options for gradient accumulation or the replication factor should mean that you do not need to set `Trainer(ipus=x, accumulate_grad_batches=x)`. These will be infered from the settings.

Reply:
We'll set this up in a separate PR if agreed on, as this still moves us towards these objectives.

Reply:
Yeah I like that

Reply:
I'll handle this in a separate PR as I feel the connector will be the bottleneck, which is being addressed slowly.